test/extended/baremetal/hosts.go

```diff
@@ -0,0 +1,68 @@
+package baremetal
+
+import (
+	"context"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	configv1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+func skipIfNotBaremetal(oc *exutil.CLI) {
+	g.By("checking platform type")
+
+	infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	if infra.Status.Platform != configv1.BareMetalPlatformType {
+		e2eskipper.Skipf("No baremetal platform detected")
+	}
+}
+
+func baremetalClient(dc dynamic.Interface) dynamic.ResourceInterface {
+	baremetalClient := dc.Resource(schema.GroupVersionResource{Group: "metal3.io", Resource: "baremetalhosts", Version: "v1alpha1"})
+	return baremetalClient.Namespace("openshift-machine-api")
+}
+
+var _ = g.Describe("[sig-installer][Feature:baremetal] Baremetal platform should", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("have a metal3 deployment", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		_, err = c.AppsV1().Deployments("openshift-machine-api").Get(context.Background(), "metal3", metav1.GetOptions{})
```

Comment:
seems like you'd want to do more here with the return.  Should it have ready replicas?

Reply:
It's a first basic test to verify that CBO (started to) work properly and (at least) deployed a `metal3` deployment, and there aren't [replicas set](https://github.com/openshift/cluster-baremetal-operator/blob/1f7ff0a7150948440fdbe6bda028de411e743a9d/provisioning/baremetal_pod.go#L632). I expect that anyhow in future this test could be slightly enriched.

Reply:
> It's a first basic test to verify that CBO (started to) work properly and (at least) deployed a `metal3` deployment, and there aren't [replicas set](https://github.com/openshift/cluster-baremetal-operator/blob/1f7ff0a7150948440fdbe6bda028de411e743a9d/provisioning/baremetal_pod.go#L632). I expect that anyhow in future this test could be slightly enriched.

it sounds like that is exactly the case you want to catch and fail on this test.  What good is the deployment without any ready replicas?

Reply:
Added a check to verify that the replica is set to one